connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -100,20 +102,39 @@ public Worker(
         this.config = config;
         this.workerMetricsGroup = new WorkerMetricsGroup(metrics);
 
-        // Internal converters are required properties, thus getClass won't return null.
-        this.internalKeyConverter = plugins.newConverter(
-                config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName(),
-                config
-        );
+        // Internal key/value converters are currently deprecated, instead JsonConverter is used by default, refer KIP-174.
+        // Currently Internal converters are required properties, thus getClass won't return null.
+
+        boolean isJsonConverterKeyClassUsed = config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+        boolean isJsonConverterValueClassUsed = config.getClass(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG).getName().equals("org.apache.kafka.connect.json.JsonConverter");
+
+        if (!isJsonConverterKeyClassUsed && !isJsonConverterValueClassUsed) {
+            this.internalKeyConverter =
+                    plugins.newConverter(config.getClass(WorkerConfig.INTERNAL_KEY_CONVERTER_CLASS_CONFIG).getName(), config);
+            this.internalValueConverter =
+                    plugins.newConverter(config.getClass(WorkerConfig.INTERNAL_VALUE_CONVERTER_CLASS_CONFIG).getName(), config);
+        } else {
+            this.internalKeyConverter = plugins.newConverter(JsonConverter.class.getName(), config);
+            this.internalValueConverter = plugins.newConverter(JsonConverter.class.getName(), config);
+        }
+
+        boolean isInternalConverterSchemaConfigured = config.originals().containsKey("internal.key.converter.schemas.enable") && config.originals().containsKey("internal.value.converter.schemas.enable");
+
+        Map<String, String> userConfigs = config.originalsStrings();
```

Comment:
I'm not sure the handling of configs here is quite right. I think it would make more sense to check if there are *any* properties set with the prefix `internal.key.converter` (and `internal.value.converter` for that case). There are other configs that could be overridden and really at this point, if the user has specified *anything* explicitly, we want to still respect those settings. Only after the deprecation period should there be cases where we might ignore/modify settings they explicitly specify.

In other words, I'm not sure a version of this code that does not maintain use of `config.originalsWithPrefix("internal.key.converter")` and `config.originalsWithPrefix("internal.value.converter")` can properly maintain compatibility.

Reply:
Agree, however I am not pretty sure if we have more configs other than `internal.key.converter.schemas.enable` and `internal.value.converter.schemas.enable` which qualify the prefix search using `config.originalsWithPrefix("internal.key.converter.")` and `config.originalsWithPrefix("internal.value.converter.")`. That is why I used the full name of these configs instead of filtering them with prefix. 

Is there any existing config which has `internal.key|value.converter.` as prefix and we need to take care of that in Worker's context ?